ravenframework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,464 @@
+      @ In, par, np.array, the parameters (inputs) of the mock experiment
+      @ In, UparVar, np.array, variance covariance matrix of the parameters in the error space
+      @ In, Umes, np.array, the error in measurements
+      @ In, UmesVar, np.array, variance covariance matrix of the measurables in the error space
+      @ In, normalizedSen, np.array, the normalized sensitivity matrix
+      @ Out, parTilde, np.array, the corrected parameters
+      @ Out, parTildeVar, np.array, the variance covariance matrix of the corrected parameters (uncertainty in the corrected parameters)
+    """
+    # Compute adjusted par #eq 48
+    UparTilde = UparVar @ normalizedSen.T @ np.linalg.pinv(normalizedSen @ UparVar @ normalizedSen.T + UmesVar) @ Umes
+
+    # back transform to parameters
+    parTilde = UparTilde * par + par
+
+    # Compute adjusted par_var #eq 67
+    UparVarTilde = UparVar - UparVar @ normalizedSen.T @ np.linalg.pinv(normalizedSen @ UparVar @ normalizedSen.T + UmesVar) @ normalizedSen @ UparVar
+
+    # back transform the variance
+    UparVarTildeDiag = np.diagonal(UparVarTilde)
+    for ind,c in enumerate(UparVarTildeDiag):
+        if c<0:
+            UparVarTilde[ind,ind] = 0
+    UparVarTildeDiag2 = np.sqrt(UparVarTildeDiag)
+    UparVarTildeDiag3 = UparVarTildeDiag2 * np.squeeze(par)
+    parVarTilde = np.square(UparVarTildeDiag3)
+    parVarTilde = np.diag(parVarTilde)
+    return parTilde, parVarTilde
+
+  def _targetCorrection(self, FOMs, UparVar, Umes, UmesVar, normalizedSenTar, normalizedSenExp):
+    """
+      A utility function to compute corrections in targets based on the representativity analysis
+      @ In, FOMs, np.array, target out puts (Figures of merit)
+      @ In, UparVar, np.array, np.array, variance covariance matrix of the parameters in the error space
+      @ In, Umes, np.array, the error in measurements
+      @ In, UmesVar, np.array, variance covariance matrix of the measurables in the error space
+      @ In, normalizedSenTar, np.array, normalized sensitivities of the target outputs w.r.t. parameterts
+      @ In, normalizedSenExp, np.array, normalized sensitivities of the mock prototype/experiment outputs (measurements) w.r.t. parameterts
+      @ Out, tarTilde, np.array, corrected targets (FOMs)
+      @ Out, tarVarTilde, np.array, variance covariance matrix for the corrected targets
+      @ Out, UtarVarTilde,  np.array, variance covariance matrix for the corrected targets in error space
+      @ Out, UtarVartilde_no_UmesVar, np.array, variance covariance matrix for the corrected targets in error space assuming no uncer
+      @ Out, propagetedExpUncert, np.array, propagated variance covariance matrix of experiments due to parameter uncertainties
+    """
+    # Compute adjusted target #eq 71
+    UtarTilde = normalizedSenTar @ UparVar @ normalizedSenExp.T @ np.linalg.pinv(normalizedSenExp @ UparVar @ normalizedSenTar.T + UmesVar) @ Umes
+    # back transform to parameters
```

Comment:
This equation might have a bug. doesn't match equation 71 in the writeup

Reply:
Is this comment resolved?